bin/propolis-server/src/lib/spec/api_spec_v0.rs

```diff
@@ -68,6 +68,7 @@ impl From<Spec> for InstanceSpecV0 {
         let mut spec = InstanceSpecV0::default();
         spec.devices.board = val.board;
         for (disk_name, disk) in val.disks {
+            let backend_name = disk.device_spec.backend_name().to_owned();
```

Comment:
it occurs to me just now that, since backend and device names never get mutated once they're parsed, we _could_ switch to using `Arc<str>` for backend names so that we don't memcpy the whole string and instead just increment its ref count. but, I don't know if this is worth the effort; each propolis-server process only has one spec, so the device/backend names probably don't represent a huge amount of memory, and we only do this on initialization, so we're not making a ton of clones in a hot loop...

Reply:
Agreed--I don't think this is likely to be performance-critical. This conversion direction (internal spec to external wire-format spec) should only really be used during live migration, when the source has to rehydrate a wire spec to send to the target; that allocation should be transient, and all this work happens at the very beginning of migration (well before blackout), so probably doesn't need to be the very fastest thing in the world.